rfd/0166-tpm-joining.md

```diff
@@ -0,0 +1,390 @@
+---
+authors: @strideynet (noah@goteleport.com)
+state: draft
+---
+
+# RFD 0166 - TPM Joining
+
+## Required Approvers
+ 
+* Engineering: (@zmb33 || @codingllama)
+* Security: (@reedloden || @jentfoo)
+* Product: (@xinding33 || @klizhentas)
+
+# What
+
+Introduce a new join method that allows a Bot or Agent running on a host with a
+TPM to securely join a Teleport cluster without using a shared secret.
+
+# Why
+
+Bootstrapping trust with a newly provisioned host in an on-premises environment
+is challenging. In many environments, this is done by transferring an initial
+shared secret to the host, which can then be used for authentication. However,
+this is difficult to complete securely - especially at scale. This is due to the
+risk of impersonation/interception, an attacker who has sufficiently compromised
+the network can impersonate a newly provisioned host and receive access to this
+secret. In addition, a shared secret on the host is liable to exfiltration.
+
+A TPM provides a secure, unique and persistent initial identity ideal for
+bootstrapping trust with a host. Even across reboots, or reconfigurations, the
+TPM identity remains the same. The guarantees provided by a compliant TPM mean
+that this identity cannot be exfiltrated. This makes it a strong candidate for
+bootstrapping trust.
+
+# Implementation
+
+Reference materials:
+
+- [TCG Credential Profile EK 2.0](https://trustedcomputinggroup.org/wp-content/uploads/TCG-EK-Credential-Profile-V-2.5-R2_published.pdf): For specification of the EK and EKCert.
+- [TCG Trusted Platform Module Library](https://trustedcomputinggroup.org/resource/tpm-library-specification/): For specification of TPM commands and structures.
+- [TPM 2.0 Keys for Device identity and Attestation](https://trustedcomputinggroup.org/wp-content/uploads/TPM-2p0-Keys-for-Device-Identity-and-Attestation_v1_r12_pub10082021.pdf): For the recommended TPM 2.0 flows for device identity and attestation.
+- [RFD 0008e - Device Trust TPM Support](https://github.com/gravitational/teleport.e/blob/master/rfd/0008e-device-trust-tpm.md): For the existing TPM identity implementation in Teleport.
+
+Existing code and logic from Device Trust TPM support will be reused where
```

Comment:
> will be reused

I'd guess more copied or used as a basis than reused. The features have different concerns and UX goals.

Reply:
Yeah - there may be a few helpers I could extract but likely mostly copy paste/referencing.